File: lib/widgets/stats/stat_card.dart
import 'package:flutter/material.dart';
import '../../../theme/app_theme.dart';

/// Enum for different stat card layout styles
enum StatCardStyle {
  /// Full-size card with icon, value, and title stacked vertically
  kpi,

  /// Compact fixed-size card (105x90) with vertical layout
  compactVertical,

  /// Compact header card (140x100) with horizontal icon + title and value below
  compactHeader,
}

/// Unified stat card widget supporting multiple layout styles
class StatCard extends StatelessWidget {
  final String title;
  final String value;
  final IconData icon;
  final Color color;
  final StatCardStyle style;

  const StatCard({
    super.key,
    required this.title,
    required this.value,
    required this.icon,
    required this.color,
    this.style = StatCardStyle.kpi,
  });

  @override
  Widget build(BuildContext context) {
    final card = _buildCardContent(context);

    // Apply size constraints if needed
    if (style == StatCardStyle.compactVertical) {
      return SizedBox(width: 105, height: 90, child: card);
    } else if (style == StatCardStyle.compactHeader) {
      return SizedBox(width: 140, height: 100, child: card);
    }

    return card;
  }

  Widget _buildCardContent(BuildContext context) {
    return Card(
      elevation: 0,
      shape: RoundedRectangleBorder(
        side: BorderSide(color: color, width: AppTheme.neoBorderWidth),
        borderRadius: AppTheme.brNeo,
      ),
      color: color.withValues(alpha: 0.08),
      child: Padding(padding: _getPadding(), child: _buildContent(context)),
    );
  }

  EdgeInsetsGeometry _getPadding() {
    return switch (style) {
      StatCardStyle.kpi => const EdgeInsets.all(16.0),
      StatCardStyle.compactVertical => const EdgeInsets.all(
        AppTheme.spaceSmall,
      ),
      StatCardStyle.compactHeader => const EdgeInsets.all(AppTheme.spaceMedium),
    };
  }

  Widget _buildContent(BuildContext context) {
    return switch (style) {
      StatCardStyle.kpi => _buildKpiContent(context),
      StatCardStyle.compactVertical => _buildCompactVerticalContent(context),
      StatCardStyle.compactHeader => _buildCompactHeaderContent(context),
    };
  }

  Widget _buildKpiContent(BuildContext context) {
    return Column(
      crossAxisAlignment: CrossAxisAlignment.start,
      children: [
        Icon(icon, color: color),
        const SizedBox(height: 12),
        Text(
          value,
          style: const TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
        ),
        Text(
          title,
          style: TextStyle(
            fontSize: 12,
            color: Theme.of(context).colorScheme.onSurfaceVariant,
          ),
        ),
      ],
    );
  }

  Widget _buildCompactVerticalContent(BuildContext context) {
    return Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [
        Icon(icon, color: color, size: AppTheme.iconSizeSmall),
        const Spacer(),
        Text(
          value,
          style: const TextStyle(
            fontSize: AppTheme.fsHeadingMedium,
            fontWeight: AppTheme.fwExtraBold,
          ),
        ),
        Text(
          title,
          textAlign: TextAlign.center,
          style: TextStyle(
            fontSize: AppTheme.fsCaption,
            color: Theme.of(context).colorScheme.onSurfaceVariant,
            fontWeight: AppTheme.fwBold,
          ),
        ),
      ],
    );
  }

  Widget _buildCompactHeaderContent(BuildContext context) {
    return Column(
      mainAxisAlignment: MainAxisAlignment.center,
      children: [
        Row(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Icon(icon, color: color, size: AppTheme.iconSizeSmall),
            const SizedBox(width: AppTheme.spaceXSmall),
            Text(
              title,
              style: TextStyle(
                fontSize: AppTheme.fsCaption,
                color: Theme.of(context).colorScheme.onSurfaceVariant,
                fontWeight: AppTheme.fwBold,
              ),
            ),
          ],
        ),
        const Spacer(),
        Text(
          value,
          style: const TextStyle(
            fontSize: AppTheme.fsHeadingMedium,
            fontWeight: AppTheme.fwExtraBold,
          ),
        ),
      ],
    );
  }
}

File: app/src/main/java/io/github/benji377/timety/ui/components/stats/StatCard.kt
package io.github.benji377.timety.ui.components.stats

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import io.github.benji377.timety.ui.theme.AppTheme


/** Layout variants for [StatCard]. */
enum class StatCardStyle { KPI, COMPACT_VERTICAL, COMPACT_HEADER }


/** Bordered card showing an icon, [value], and [title]; [style] picks the layout and sizing. */
@Composable
fun StatCard(
    title: String,
    value: String,
    icon: ImageVector,
    color: Color,
    modifier: Modifier = Modifier,
    style: StatCardStyle = StatCardStyle.KPI,
) {
    val sized = when (style) {
        StatCardStyle.COMPACT_VERTICAL -> modifier.size(width = 105.dp, height = 90.dp)
        StatCardStyle.COMPACT_HEADER -> modifier.size(width = 140.dp, height = 100.dp)
        StatCardStyle.KPI -> modifier
    }

    Card(
        modifier = sized,
        shape = AppTheme.brNeo,
        border = BorderStroke(AppTheme.neoBorderWidth, color),
        colors = CardDefaults.cardColors(containerColor = color.copy(alpha = 0.08f)),
        elevation = CardDefaults.cardElevation(defaultElevation = 0.dp),
    ) {
        when (style) {
            StatCardStyle.KPI -> KpiContent(title, value, icon, color)
            StatCardStyle.COMPACT_VERTICAL -> CompactVerticalContent(title, value, icon, color)
            StatCardStyle.COMPACT_HEADER -> CompactHeaderContent(title, value, icon, color)
        }
    }
}

@Composable
private fun KpiContent(title: String, value: String, icon: ImageVector, color: Color) {
    Column(modifier = Modifier.padding(16.dp)) {
        Icon(icon, contentDescription = null, tint = color)
        Spacer(Modifier.height(12.dp))
        Text(value, fontSize = 24.sp, fontWeight = AppTheme.fwBold)
        Text(
            title,
            fontSize = 12.sp,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
        )
    }
}

@Composable
private fun CompactVerticalContent(title: String, value: String, icon: ImageVector, color: Color) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(AppTheme.spaceSmall),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        Icon(
            icon,
            contentDescription = null,
            tint = color,
            modifier = Modifier.size(AppTheme.iconSizeSmall)
        )
        Spacer(Modifier.height(4.dp))
        Text(
            value,
            fontSize = AppTheme.fsHeadingMedium,
            fontWeight = AppTheme.fwExtraBold,
            maxLines = 1,
            softWrap = false,
            overflow = TextOverflow.Ellipsis,
        )
        Spacer(Modifier.height(4.dp))
        Text(
            title,
            textAlign = TextAlign.Center,
            maxLines = 2,
            overflow = TextOverflow.Ellipsis,
            fontSize = AppTheme.fsCaption,
            color = MaterialTheme.colorScheme.onSurfaceVariant,
            fontWeight = AppTheme.fwBold,
        )
    }
}

@Composable
private fun CompactHeaderContent(title: String, value: String, icon: ImageVector, color: Color) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .padding(AppTheme.spaceMedium),
        verticalArrangement = Arrangement.Center,
    ) {
        Row(verticalAlignment = Alignment.Top) {
            Icon(
                icon,
                contentDescription = null,
                tint = color,
                modifier = Modifier.size(AppTheme.iconSizeSmall)
            )
            Spacer(Modifier.width(AppTheme.spaceXSmall))
            Text(
                title,
                maxLines = 2,
                overflow = TextOverflow.Ellipsis,
                fontSize = AppTheme.fsCaption,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                fontWeight = AppTheme.fwBold,
                modifier = Modifier.weight(1f),
            )
        }
        Spacer(Modifier.weight(1f))
        Text(
            value,
            fontSize = AppTheme.fsHeadingMedium,
            fontWeight = AppTheme.fwExtraBold,
            maxLines = 1,
            softWrap = false,
            overflow = TextOverflow.Ellipsis,
        )
    }
}
